common/remote/hooks/git/gitfile.cpp

```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+    {
+        throwUnexpected();
+    }
+    virtual offset_t appendFile(IFile *file,offset_t pos=0,offset_t len=(offset_t)-1)
+    {
+        throwUnexpected();
+    }
+    virtual void setSize(offset_t size)
+    {
+        throwUnexpected();
+    }
+    virtual void flush()
+    {
+        throwUnexpected();
+    }
+    unsigned __int64 getStatistic(StatisticKind kind)
+    {
+        //This could be implemented, but not likely to be useful so currently return nothing.
+        return 0;
+    }
+protected:
+    MemoryBuffer buf;
+};
+
+class NewGitRepositoryFile : public GitRepositoryFile
+{
+public:
+    NewGitRepositoryFile(const char *_gitFileName, offset_t _fileSize, bool _isDir, bool _isExisting, GitCommitTree * _commitTree, const git_oid * _oid)
+    : GitRepositoryFile(_gitFileName, _fileSize, _isDir, _isExisting), commitTree(_commitTree), oid(*_oid)
+    {
+    }
+
+    virtual IFileIO * open(IFOmode mode, IFEflags extraFlags) override
+    {
+        assertex(mode==IFOread && isExisting);
+        return new NewGitRepositoryFileIO(gitDirectory, revision, relFileName, commitTree, &oid);
+    }
+    virtual IFileIO * openShared(IFOmode mode, IFSHmode shmode, IFEflags extraFlags) override
+    {
+        assertex(mode==IFOread && isExisting);
+        return new NewGitRepositoryFileIO(gitDirectory, revision, relFileName, commitTree, &oid);
+    }
+
+protected:
+    Linked<GitCommitTree> commitTree;
+    const git_oid oid;
+};
+class NewGitRepositoryDirectoryIterator : implements IDirectoryIterator, public CInterface
+{
+    static int treeCallback(const char *root, const git_tree_entry *entry, void *payload)
+    {
+        NewGitRepositoryDirectoryIterator * self = reinterpret_cast<NewGitRepositoryDirectoryIterator *>(payload);
+        return self->noteEntry(root, entry);
+    }
+public:
+    IMPLEMENT_IINTERFACE;
+    NewGitRepositoryDirectoryIterator(const char *_gitFileName, const char *_mask, bool _sub, bool _includeDirs)
+    : mask(_mask), sub(_sub), includeDirs(_includeDirs)
+    {
+        splitGitFileName(_gitFileName, gitDirectory, revision, relDir);
+        curIndex = 0;
+
+        const char * version = revision.length() ? revision.get() : "HEAD";
+        commitTree.setown(new GitCommitTree(gitDirectory, version));
+        if (!commitTree->queryTree())
+            throw makeStringExceptionV(9900 , "Cannot resolve git revision %s", _gitFileName);
+    }
+    ~NewGitRepositoryDirectoryIterator()
+    {
+    }
+    virtual StringBuffer &getName(StringBuffer &buf)
+    {
+        assertex(curFile);
+        return buf.append(curFile->queryFilename());
+    }
+    virtual bool isDir()
+    {
+        assertex(curFile);
+        return curFile->isDirectory()==fileBool::foundYes;
+    }
+    virtual __int64 getFileSize()
+    {
+        assertex(curFile);
+        return curFile->size();
+    }
+    virtual bool getModifiedTime(CDateTime &ret)
+    {
+        UNIMPLEMENTED;
```

Comment:
Will it be?

Reply:
It wasn't implemented in the previous code - so probably not unless it is needed.